fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -710,16 +765,21 @@ private SqlQueryData buildChainedIdClause(QueryParameter currentParm, String cha
 
     private void appendMidChainParm(StringBuilder whereClauseSegment, QueryParameter currentParm, String chainedParmVar)
             throws FHIRPersistenceDBConnectException, FHIRPersistenceDataAccessException, FHIRPersistenceException {
-        Integer parameterNameId = ParameterNamesCache.getParameterNameId(currentParm.getCode());
-        if (parameterNameId == null) {
-            parameterNameId = this.parameterDao.readParameterNameId(currentParm.getCode());
-            if (parameterNameId != null) {
-                this.parameterDao.addParameterNamesCacheCandidate(currentParm.getCode(), parameterNameId);
+        Integer parameterNameId = identityCache.getParameterNameId(currentParm.getCode());
+        whereClauseSegment.append(chainedParmVar).append(DOT).append("PARAMETER_NAME_ID").append(EQ)
+                .append(parameterNameId);
+
+        // TODO handle code system lookup failures here. Perhaps = NULL to make the query return no rows?
+        final String codeSystemName = currentParm.getModifierResourceTypeName();
+        if (codeSystemName != null && !codeSystemName.equals("*")) {
```

Comment:
Repetition... maybe a helper method to manage?

Reply:
Too risky to change anything here at this point, but I agree a helper would be useful. Raising an issue for the backlog.